spec/views/cases/filters/index_html_slim_spec.rb

```diff
@@ -0,0 +1,273 @@
+    expect(second_case.number.text).to eq "Case number #{assigned_case.number}"
+    expect(second_case.type.text).to eq "FOI "
+    expect(second_case.flag.text).to eq "This is a Trigger case"
+    expect(second_case.request_detail.text)
+      .to eq "#{ assigned_case.subject } #{assigned_case.name}"
+    expect(second_case.draft_deadline.text).to eq assigned_case.internal_deadline
+    expect(second_case.external_deadline.text)
+      .to eq assigned_case.external_deadline
+    expect(second_case.status.text).to eq assigned_case.status
+    expect(second_case.who_its_with.text).to eq assigned_case.who_its_with
+
+    third_case = cases_page.case_list[3]
+    expect(third_case.number.text).to eq "Case number #{accepted_case.number}"
+    expect(third_case.request_detail.text)
+      .to eq "#{ accepted_case.subject } #{accepted_case.name}"
+    expect(third_case.draft_deadline.text).to eq accepted_case.internal_deadline
+    expect(third_case.external_deadline.text)
+      .to eq accepted_case.external_deadline
+    expect(third_case.status.text).to eq accepted_case.status
+    expect(third_case.who_its_with.text).to eq accepted_case.who_its_with
+
+    fourth_case = cases_page.case_list[4]
+    expect(fourth_case.number.text).to eq "Case number #{ovt_foi_trigger_case.number}"
+    expect(fourth_case.type.text).to eq "ICO overturned (FOI) "
+    expect(fourth_case.request_detail.text)
+        .to eq "#{ovt_foi_trigger_case.subject} #{ovt_foi_trigger_case.name}"
+    expect(fourth_case.draft_deadline.text).to eq ovt_foi_trigger_case.internal_deadline
+    expect(fourth_case.external_deadline.text)
+        .to eq ovt_foi_trigger_case.external_deadline
+    expect(fourth_case.status.text).to eq ovt_foi_trigger_case.status
+    expect(fourth_case.who_its_with.text).to eq ovt_foi_trigger_case.who_its_with
+
+    fifth_case = cases_page.case_list[5]
+    expect(fifth_case.number.text).to eq "Case number #{unflagged_case.number}"
+    expect(fifth_case.type.text).to eq "FOI "
+    expect(fifth_case.request_detail.text)
+      .to eq "#{unflagged_case.subject} #{unflagged_case.name}"
+    expect(fifth_case.draft_deadline.text).to eq ' '
+    expect(fifth_case.external_deadline.text)
+      .to eq unflagged_case.external_deadline
+    expect(fifth_case.status.text).to eq unflagged_case.status
+    expect(fifth_case.who_its_with.text).to eq unflagged_case.who_its_with
+  end
+
+  it 'displays the offender SAR related cases given it' do
+    login_as branston_responder
+    offender_sar_case
+    offender_sar_case_third_party_name
+    offender_sar_case_third_party_company
+
+    assign(:cases, PaginatingDecorator.new(Case::Base.all.page.order(:number)))
+    assign(:state_selector, StateSelector.new( {} ))
+    assign(:query, search_query)
+    assign(:action_url, '/cases/open')
+    assign(:current_tab_name, 'open')
+
+    disallow_case_policy :can_add_case?
+
+    render
+    cases_page.load(rendered)
+
+    zeroth_case = cases_page.case_list[0]
```

Comment:
same here would be able to reuse that `case_expectations` method

Reply:
@willmcb  thanks for this advice 👍 ,  I added 2 function to compare those fields,